### PR TITLE
Initial attempt at connector pool handler

### PR DESCRIPTION
**Pull request by sebito91**
This is still pending a lot of work, please just consider that when looking at code.

### CODE REVIEW

File: connection/connection.go [hunk 1]
```diff
@@ -5,10 +5,12 @@ type ID string
...
 // Connection is a configuration used by the connector to transport data from source to the EG (target).
 type Connection struct {
-	Space  string           `json:"space"`
-	ID     ID               `json:"connectionId"`
-	Target string           `json:"target"`
-	Source AWSKinesisSource `json:"source"`
+	Space            string            `json:"space"`
+	ID               ID                `json:"connectionId"`
+	Target           string            `json:"target"`
+	Type             string            `json:"type"`
```
Comment (mthenw) [inline connection/connection.go]: In EG it's called EventType and we should be consistent

File: connection/connection.go [hunk 2]
```diff
@@ -5,10 +5,12 @@ type ID string
...
+	AWSKinesisSource *AWSKinesisSource `json:"AWSKinesisSource,omitempty"`
```
Comment (mthenw) [inline connection/connection.go]: We should follow convention from EG here https://github.com/serverless/event-gateway/blob/master/docs/api.md#register-function

There should be a single "source" field that depending on "type" have a different structure. Also, as it's PoC I don't think it's a priority right now.

File: connection/connection.go [hunk 3]
```diff
@@ -19,3 +21,11 @@ type AWSKinesisSource struct {
 	AWSSecretAccessKey string `json:"awsSecretAccessKey,omitempty"`
 	AWSSessionToken    string `json:"awsSessionToken,omitempty"`
 }
+
+// KafkaSource is the configuration struct to leverage when defining a Kafka resource
+// TODO: add tls config portion
+type KafkaSource struct {
```
Comment (mthenw) [inline connection/connection.go:27]: Let's focus the initial pass on Kinesis.

File: cmd/pool/main.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package main
```
Comment (mthenw) [inline cmd/pool/main.go]: I'm not following. Two commands? What's the reason for another command?

File: pool/workers.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+// StartWorkers receives the number of worker goroutines from the main process
+func StartWorkers(numWorkers int, conns chan *connection.Connection, done <-chan bool) error {
+	// initialize the logger for the pool
+	rawLogger, _ := zap.NewDevelopment()
+	defer rawLogger.Sync()
+	log := rawLogger.Sugar()
+
+	m := make(workerMap)
```
Comment (mthenw) [inline pool/workers.go:27]: what's `m`?

File: pool/workers_test.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+	"github.com/serverless/event-gateway-connector/connection"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestInitialWorkerConnection(t *testing.T) {
+	conns := make(chan *connection.Connection, 100)
+	done := make(chan bool)
+
+	Convey("test out initializing the workers", t, func() {
```
Comment (mthenw) [inline pool/workers_test.go:14]: At this point I don't think it's needed.

**Comment by mthenw** [comment]
@sebito91 I've done some initial review.

We also need to add the number of workers per connection. It will change the approach a bit so I think it's best to do as early as possible.

**Comment by sebito91** [comment]
Thanks, this was the initial cut to get started. You can disregard the `cmd/pool/main.go` as that was more for testing than anything else. It will not be required in the final solution.

As for the `kafka` parts, I've only added as a stub for now to show that there would be other components. We were hardcoding the `source` to `AWSKinesisSource` values and wanted to show that it could be different. I agree that we should align these and make them into one field, will need to think more about how to do that.

In any case, thanks for the initial feedback. This is still WIP so please don't consider this yet as final product.

File: cmd/pool/main.go (hunk 4 above)
Comment (sebito91) [reply cmd/pool/main.go]: Not going to persist, this was there for me to test with. I'll be removing with the final instance later.

File: connection/connection.go (hunk 3 above)
Comment (sebito91) [reply connection/connection.go:27]: Yes, just wanted to add color that there would be additional source types later. We were making `Source` previously into `AWSKinesis` stuff only.

File: connection/connection.go (hunk 2 above)
Comment (sebito91) [reply connection/connection.go]: Thanks for the tip on this one, was thinking about how we could do this. I like your approach in setting a custon `marshal/unmarshal` based on the type. Will update.

**Comment by sebito91** [comment]
Need to move this out from a fork

**Pull request closed without merging**